### PR TITLE
[case](regresscases) add regress cases for nested type nested type with csv format

### PR DESCRIPTION
**Pull request by amorynan**
## Proposed changes

Issue Number: close #xxx
this pr
1.  fix use podarray push_back() with back() will make heap_use_after_free when podarray is reach capacity which would may make heap free 
2. add cases for csv format for nested types. and csv file has two define which are without quote or just like json text

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

**Comment by amorynan** [comment]
run buildall

File: be/src/vec/data_types/serde/data_type_array_serde.cpp [hunk 1]
```diff
@@ -94,7 +94,8 @@ Status DataTypeArraySerDe::deserialize_one_cell_from_json(IColumn& column, Slice
     }
     // empty array []
     if (slice.size == 2) {
-        offsets.push_back(offsets.back());
+        auto last_off = offsets.back();
```
Comment (xiaokang) [inline be/src/vec/data_types/serde/data_type_array_serde.cpp:97]: fix similar issues in another pr

**Review by xiaokang — APPROVED** [review]
LGTM

**Comment by amorynan** [comment]
run p0

**Comment by amorynan** [comment]
run buildall

**Review by morningman — APPROVED** [review]
lgtm

**Pull request merged**